### DISCUSSION TITLE
Query to have the latest albums inserted in the database

### DISCUSSION THREAD

**Discussion by omersorrenti76** [MusicPlayerDaemon/MPD#1442] [Q&A]
Hello,
I need to have information on the latest albums entered in the system.
For example I copied some albums to my NAS on February 10, 2022 and i have updated mpd (mpd update) 
Now if I launch the command:
**find "(modified-since \" 2022-02-10T00: 00: 00Z \ ")"** returns nothing. 
I probably have to query using Last-Modified but I don't understand how.
Is there a way to find the latest albums added to mpd?

Can you help me? thank you very much

**Comment by MaxKellermann** [comment]
So... what is the modification time of the files you expect to be returned?

**Reply by omersorrenti76** [reply]
hello,
you probably have already implemented the solution. It is I who cannot use it.
I don't know if the correct way is to use
modified-since (but in this case it doesn't seem to be returning the albums posted after midnight on February 10th.
Or run a query using Last-Modified but I couldn't build it correctly, in case if you can kindly send me an example. 
My MPD is  MPD 0.21.4.
I thank you so much

**Reply by MaxKellermann** [reply]
MPD 0.21.4 is unsupported. See https://www.musicpd.org/help/

**Reply by omersorrenti76** [reply]
I'm sorry I hadn't seen.
I will immediately update MPD
Once updated the correct command will then be modified-since?
thank you so much

**Reply by MaxKellermann** [reply]
modified-since compares the file modification time, not when the files were added to MPD.

**Reply by omersorrenti76** [reply]
OK, thanks a lot.
So what do you recommend to get the latest albums added?
to use find with last-modified?

**Reply by MaxKellermann** [reply]
MPD does not remember when a music file was first found, so you can't do that. You can use last-modified, which refers to the file's modification time stamp.

**Reply by omersorrenti76** [reply]
good,
thank you for your time.

**Comment by slystone** [comment]
I use two bash scripts to work around the problem.

One creates the playlists with the files added in the last 24 hours and in the last week. I run the script manually:

**last_pls.sh**

```
#!/bin/bash

find    \
        -L /var/lib/mpd/music/NAS/*LOSSLESS                             \
        -mtime -7                                                       \
        \( -iname "*.flac" -o -iname "*.mp3" \) \
        | cut -d '/' -f6- | sort > /var/lib/mpd/playlists/THIS_WEEK.m3u


find    \
        -L /var/lib/mpd/music/NAS/*LOSSLESS                             \
        -mtime -1                                                       \
        \( -iname "*.flac" -o -iname "*.mp3" \) \
        | cut -d '/' -f6- | sort > /var/lib/mpd/playlists/TODAY.m3u
```

The other script updates mpd only on the latest added folders, without having to check the entire music library. I run it everyday via crontab

**today_update.sh**

```
#!/bin/bash

MUSIC_PATH=/var/lib/mpd/music
LAST_DAYS=1

cd "$MUSIC_PATH" && \
find    -L . \
        -maxdepth 3 \
        -mindepth 3 \
        -type d \
        -mtime "-${LAST_DAYS}" \
  -printf "\"%P\" " | xargs mpc --wait update
```

I hope I can be of help to you.

**Reply by guestguestguestguest** [reply]
Thanks for the scripts; I adapted last_pls.sh for my mpd server.  Good idea.

**Comment by jcorporation** [comment]
With mpd 0.24 (not yet released) there is a new added timestamp.
